source/include/TDCodecVia.h

```diff
@@ -207,6 +207,31 @@ class TDViaFormatter
     static ViaFormatChars formatC;
 };
 
+template <class T>
+class ExpandableStaticArray
+{
+ private:
+    IntIndex _staticLength;
+    T* _staticArray;
+    std::vector<T> _dynamicArray;
+
+ public:
+    ExpandableStaticArray(T* array, Int32 length) {
+        _staticArray = array;
+        _staticLength = length;
+    }
+    T& operator[] (Int32 index) {
+        if (index < _staticLength)
+            return _staticArray[index];
+        index -= _staticLength;
+        if (index >= _dynamicArray.size())
+            _dynamicArray.resize(index + 1);
```

Comment:
Consider having a chunk size by which you grow, and call _dynamicArray.reserve(index + kChunkSize) so the array doesn't have to resize and realloc memory on every single addition past the static size.

Reply:
Also, consider making the API mimic like std::vector.

Reply:
Do you mean by including methods like resize and begin and end? Is the main benefit of wrapping this in our own class over just using std::vector that we can pre-allocate the "usual" amount needed in a stack allocated array?

Reply:
It's not so much the names of the API entries that I'm concerned about, it's that the functionality is self-contained and it seems like a usable data structure on its own.   Holding onto a static array but not knowing how many elements are actually used seems half-done.   Making it use something like push_back rather than auto-inflating when indexing the element would be better.  And since it is a generic data structure and not something very Vireo specific, adopting the STL convention rather than using names that makes it look like it could be a Vireo TypedArray seems reasonable to me, but I don't feel strongly about the names.  If you want to keep Clear() and Length(), and add AddElement() (or something) instead of push_back(), that's fine with me.

Reply:
Sounds good. I've updated the helper class to take a size_t template argument and added the push_back. I also changed the []operator to no longer allocate more space, since push_back should be used in that kind of scenario.